libpod/kube.go

```diff
@@ -1153,7 +1153,7 @@ func generateKubeSecurityContext(c *Container) (*v1.SecurityContext, bool, error
 	allowPrivEscalation := !c.config.Spec.Process.NoNewPrivileges
 
 	var capabilities *v1.Capabilities
-	if !privileged {
+	if !privileged && c.config.Spec.Process.Capabilities != nil {
 		// Running privileged adds all caps.
 		capabilities = c.capAddDrop(c.config.Spec.Process.Capabilities)
```

Comment:
Should we move the check to c.capAddDrop and just check for nil there?

Reply:
That also works and is probably more readable.